### PR TITLE
Configure FUIDAO for ApprovalRequest

### PR DESCRIPTION
**Pull request by lchanmann**
## Ref
- https://nanopay.atlassian.net/browse/NP-5805

## Changes
- Cleanup FUIDAO
- Configure FUIDAO for ApprovalRequest and update ApprovalRequest.id to string

### CODE REVIEW

File: src/foam/dao/FUIDAO.js [hunk 1]
```diff
@@ -70,12 +65,10 @@ foam.CLASS({
       javaCode: `
         var id = getPropertyInfo().get(obj);
         var klass = getPropertyInfo().getValueClass();
-        synchronized (this) {
-          if ( klass == String.class && ((String) id).isBlank() ) {
-            getPropertyInfo().set(obj, getUIDGenerator().getNextString());
-          } else if ( klass == long.class && ((long) id) == 0L ) {
-            getPropertyInfo().set(obj, getUIDGenerator().getNextLong());
-          }
+        if ( klass == String.class && ((String) id).isBlank() ) {
```
Comment (jlhughes) [inline src/foam/dao/FUIDAO.js]: why not  ! SafetyUtil.isEmpty  to also test for null? 
or 
```suggestion
        if ( klass == String.class && id != null && ((String) id).isBlank() ) {
```

File: src/foam/nanos/approval/ApprovalRequest.js [hunk 2]
```diff
@@ -166,7 +166,7 @@ foam.CLASS({
       columnPermissionRequired: true
     },
     {
-      class: 'Long',
+      class: 'String',
```
Comment (jlhughes) [inline src/foam/nanos/approval/ApprovalRequest.js:169]: Does this load existing approval requests?

Comment (lchanmann) [reply src/foam/nanos/approval/ApprovalRequest.js:169]: No, it doesn't, we would need to run the migration script.

Comment (lchanmann) [reply src/foam/nanos/approval/ApprovalRequest.js:169]: Actually, the replay works fine and the old data with long ids are loaded successfully.

**Review by jlhughes — APPROVED** [review]
